controllers/suite_test.go

```diff
@@ -138,7 +138,24 @@ var _ = Describe("Reconcile Test", func() {
 			// There should always be a "Ready" condition, regardless of Status.
 			conds := getOperator.Status.Conditions
 			Expect(conds).To(Not(BeEmpty()))
-			Expect(conds).To(ContainElement(HaveField("Type", operatorsv1alpha1.TypeReady)))
+			types := operatorsv1alpha1.GetTypes()
+			Expect(conds).To(HaveLen(len(types)))
+			for _, t := range types {
+				Expect(conds).To(ContainElement(HaveField("Type", t)))
+			}
+		})
+		It("has matching gnerations in Conditions", func() {
```

Comment:
I think there's a misspelling here, but I'm not sure if `generations` was the intended word to be spelt.
```suggestion
		It("has matching generations in Conditions", func() {
```